### ISSUE TITLE
Notifications and messages appear very delayed after Signal has not been used for a while

### ISSUE THREAD

**Issue by major-mayer** [signalapp/Signal-Android#13249]
- [x] I have searched open and closed issues for duplicates
- [x] I am submitting a bug report for existing functionality that does not work as intended
- [x] I have read https://github.com/signalapp/Signal-Android/wiki/Submitting-useful-bug-reports
- [x] This isn't a feature request or a discussion topic

----------------------------------------

### Bug description
I noticed a couple of times already, that when I wake up in the morning, check my Signal messages, I only see a part of the messages and notifications for them that have arrived during the night.
Eventually they will arrive (in the chat and as a new notification), but sometimes minutes after I have seen the first message from the night.
The can create very confusing situations, where I reply to the first message that I've seen when I woke up and asking why there is no subsequent message.
Then i switch to other apps and out of the blue a "new-old" message from the night appears in my notifications and when I click on it also in the chat.
 

### Steps to reproduce
- Not really reproducible by intent

**Actual result:** Only a part of the messages from the night show up in the chat and the notifications. The rest comes later, sometimes minutes delayed, when you close Signal.
**Expected result:** All messages should appear in the correct order. If not as a notification (I could understand that there can be delays, especially during the night) then at the very latest when you are in the Signal app.

### Screenshots
![Screenshot_20231110-101052_Signal](https://github.com/signalapp/Signal-Android/assets/17907799/c7a6b5d3-93c8-4f35-b682-46d5d6c073aa)

### Device info

**Device:** Poco F3
**Android version:** 13
**Signal version:** 6.40.2

### Link to debug log
https://debuglogs.org/android/6.39.1/81014f56507984d0cfa6e828e24a8738f7e05a11a35fcd05db48e1901651c360

**Comment by major-mayer** [comment]
Just happened again, but this time I was actively using the phone all the time.
I did not receive any notification and even when I opened the app, I saw my messages were already read, but no reply was there so I sent a message/image and then the audio appeared.
It was sent 40 minutes ago...
https://debuglogs.org/android/6.40.2/4a41896fa8fabfd473d996af93ff1b93e39445e10562383d85c603fe21170fe7

**Comment by cody-signal** [comment]
Thanks, could you provide the message timestamp for the message that appeared later? You can long press the message->info->long press Sent and that should copy the timestamp? That would help us know where to look in your logs.

Also, do you know if the other person is on iOS?

**Comment by major-mayer** [comment]
> Just happened again, but this time I was actively using the phone all the time. I did not receive any notification and even when I opened the app, I saw my messages were already read, but no reply was there so I sent a message/image and then the audio appeared. It was sent 40 minutes ago... https://debuglogs.org/android/6.40.2/4a41896fa8fabfd473d996af93ff1b93e39445e10562383d85c603fe21170fe7

So the timestamp from this message should be: 1699789445140

And from my original bug report: 1699569524782

And no, the other person is not using iOS

**Comment by major-mayer** [comment]
Happened once again, but this time with a one time image.
Delay time about 1 minute before I opened the chat and opened Signal multiple times before that.
message id 1700601138539
https://debuglogs.org/android/6.41.2/574bf5cc2c1d657bd39c4db46714d4fcc31383382a08cf81e586a5f13e14a77e

**Comment by major-mayer** [comment]
And once again this happened, but this time with plain text messages, so i really don't think that it's related to the message size.
As before, I saw a few messages and wondered why my chat partner has only replied partially, and wrote this as the answer.
More or less at the same time when I pressed the send button, the remaining text messages popped up.
My chat partner also experienced the problem, but less frequently.

Here is the debug log again: https://debuglogs.org/android/6.42.3/e1bffb6798076507fc64e2073dccd8f61bcb3d76ddd4ee8191864d5c8245d769
Should be at the end of the log (most recent), because i saved the log directly afterwards.

@cody-signal Does it make sense to post additional debug logs, or are they not really helpful in this case?
